vllm/v1/core/scheduler.py

```diff
@@ -507,23 +515,27 @@ def update_from_output(
             # Transmit partial if chunked prefill & prompt logprobs is enabled
             if new_token_ids or prompt_logprobs_tensors is not None:
                 # Add EngineCoreOutput for this Request.
-                outputs.append(
-                    EngineCoreOutput(
-                        request_id=req_id,
-                        new_token_ids=new_token_ids or [],
-                        finish_reason=request.get_finished_reason(),
-                        new_logprobs=new_logprobs,
-                        new_prompt_logprobs_tensors=prompt_logprobs_tensors,
-                        stop_reason=request.stop_reason))
+                # Add EngineCoreOutputs for this Request.
+                output.request_ids.append(req_id)
+                output.new_token_id_offsets.append(offset)
+                new_ids = request.output_token_ids[-num_new_tokens:]
+                output.new_token_ids += new_ids
+
+                # TODO: This is not right
+                output.new_logprobs += new_logprobs
+                output.new_prompt_logprobs_tensors += (prompt_logprobs_tensors if prompt_logprobs_tensors is not None else [])
```

Comment:
I think this can be done outside of the loop, just set the dict directly in the outputs (no need to get for each request in the loop)